src/ORM/ArrayList.php

```diff
@@ -604,7 +604,7 @@ public function canFilterBy($by)
 
         $firstRecord = $this->first();
 
-        return is_array($firstRecord) ? array_key_exists($by, $firstRecord) : property_exists($by, $firstRecord ?? '');
+        return is_array($firstRecord) ? array_key_exists($by, $firstRecord) : property_exists($firstRecord ?? '', $by);
```

Comment:
```suggestion
        return is_array($firstRecord) ? array_key_exists($by, $firstRecord) : property_exists($firstRecord, $by ?? '');
```
We shouldn't be ensuring the _record_ is a string, but the _property name_.

Reply:
Well, I kept the original check because it ends with an error only if the `object_or_class` parameter ($firstRecord) is null and that can happen since `$this->first()` can return null, so basically property name should not cause an issue.

Reply:
The null coallescing operator was added during an automated code update for PHP compatbility. The update found any built-in PHP function that requires a string argument and added a null coallescing operator if a variable was being passed in there. In this case, it's the property name that was affected.

Please change this so that the argument which PHP requires to be a string (the property name) has the null coallescing operator and not the object_or_class argument.

Reply:
Okay, thank you for the explanation. I will update it.